testing/hierarchies.py

```diff
@@ -79,6 +80,12 @@ def __str__(self):
         """
         return "{0}: {1}".format(self.kind, self.name)
```

Comment:
TODO: should be able to subsume `ExhaleNode.__str__` here

Reply:
decision reversed, __repr__ kept, improved it in ExhaleNode and use it in testing framework